lib/internal/test_runner/test.js

```diff
@@ -71,6 +71,8 @@ const kUnwrapErrors = new SafeSet()
   .add('uncaughtException').add('unhandledRejection');
 const { testNamePatterns, testOnlyFlag } = parseCommandLine();
 
+const completedTests = new SafeSet();
```

Comment:
I would like to get rid of this if possible. Each `Test` object now has a `harness` field that can be used to hold this "process level state." It is only used on the root test, and is currently initialized in harness.js but there is no reason it couldn't be initialized in the `Test` constructor. I have considered giving every `Test` a reference to the root test for a while now - I think that could simplify some things, including this.

We could put the `counters` object in the `harness` since it has the same scope as the other pieces of data that are stored in it. Then, instead of tracking all of the tests in a `Set`, every test could update the counters directly via `this.root.harness.counters`.

I think this would also get rid of the need for `omitFromCounters` because each test/suite would be doing its own updating (or not updating) of the counters.